Update on observability strategy (#780)
kyma-project/community: concepts/observability-strategy/target-architecture.md

## Types
### Logs

The logging domain will provide an optional agent for tailing logs from containers running on the nodes, supporting selection on namespace name. Furthermore, kubernetes events can be streamed as logs via a dedicated agent.
The gateway will provide a durable buffering if demanded which requires to run the gateway as a StatefulSet with a maximum buffer capacity.

### Metrics

For metrics the core agent will be the annotation based scraper agent which will support scraping custom metrics from your applications. On top, istio proxy and system components can be scraped.
Furthermore, the typical kubernetes agents will be supported for retrieving host metrics, kubelet metrics and scraping APIServer metrics. A receiver for metrics about the resources stored at the kubernetes APIServer will complement the setup.

The metrics gateway will not have any special feature.

### Traces

> For traces there will be no dedicated agent, instead the system components will feed the trace data. However, the gateway setup will be special. In order to support a scalable tail based sampling, a two-stage component setup is needed. First a deployment which is then detaching spans belonging to the same trace always to the same instance of the preceding StatefulSet. That will allow filtering based on full traces.

Reviewer: ```suggestion
For traces, there will be no dedicated agent; instead the system components will feed the trace data. However, the gateway setup will be special. To support scalable tail-based sampling and filtering based on full traces, all spans of a trace must be processed by the same instance of the preceding StatefulSet. We achieve this with a two-stage component setup: First a deployment, which then dispatches all spans of a trace to the same instance.
```
